crates/git/src/repository.rs

```diff
@@ -770,6 +774,78 @@ impl GitRepository for RealGitRepository {
             ));
         }
     }
+
+    fn check_for_pushed_commit(&self) -> Result<Vec<SharedString>> {
+        let working_directory = self.working_directory()?;
+
+        let output = new_std_command(&self.git_binary_path)
+            .current_dir(&working_directory)
+            .args(["rev-parse", "HEAD"])
+            .output()?;
+        if !output.status.success() {
+            return Err(anyhow!(
+                "Failed to get HEAD:\n{}",
+                String::from_utf8_lossy(&output.stderr)
+            ));
+        }
+        let head = String::from_utf8(output.stdout)?.trim().to_owned();
+
+        let output = new_std_command(&self.git_binary_path)
+            .current_dir(&working_directory)
+            .args(["remote"])
+            .output()?;
+        if !output.status.success() {
+            return Err(anyhow!(
+                "Failed to get remotes:\n{}",
+                String::from_utf8_lossy(&output.stderr)
+            ));
+        }
+
+        let mut matching_remotes = vec![];
+
+        // TODO: current upstream branch
+        for remote in String::from_utf8(output.stdout)?.lines() {
+            let output = new_std_command(&self.git_binary_path)
+                .current_dir(&working_directory)
+                .args(["symbolic-ref", &format!("refs/remotes/{remote}/HEAD")])
+                .output()?;
+            if !output.status.success() {
+                continue; // remotes that you haven't fetched from don't have this ref
+            }
+            let remote_head = String::from_utf8(output.stdout)?.trim().to_owned();
+
+            let output = new_std_command(&self.git_binary_path)
+                .current_dir(&working_directory)
+                .args(["merge-base", &head, &remote_head])
+                .output()?;
+            let merge_base = String::from_utf8(output.stdout)?.trim().to_owned();
+            if output.status.success() && merge_base == head {
+                if let Some(s) = remote_head.strip_prefix("refs/remotes/") {
+                    matching_remotes.push(s.to_owned().into());
+                }
+            }
+        }
+
+        let output = new_std_command(&self.git_binary_path)
+            .current_dir(&working_directory)
+            .args(["rev-parse", "--symbolic-full-name", "@{u}"])
+            .output()?;
+        if output.status.success() {
+            let remote_head = String::from_utf8(output.stdout)?.trim().to_owned();
+            let output = new_std_command(&self.git_binary_path)
```

Comment:
Can we de-duplicate this command and output checking code? It looks like the merge base calls are the same in both the remote and the upstream cases.